Added Japanese localization

danesky-jp opened the pull request:
Japanese localization by me (danesky) is added as a part of contributions to SJ2021, hopefully. Fonts missing in vanilla Celeste are also provided.

At present, there are multiple png files, because information of all characters in Japanese.txt is embedded there. In the future, I may reduce the number of files by removing overlapping characters, but for now this is OK to me.

danesky-jp committed:
Added Japanese localization

danesky-jp committed:
Changed file name of png files

to be consistent with existing Chinese translation

WEGFan commented:
You can only enable the localization "mod" but without the fonts and launch the game, then you can type `check_language japanese` in the game console to get a list of missing characters, it will be written in `japanese-missing-debug.txt` in your game installation folder:

```
含項並節未類@抽証警複共布列依
```

*The following is not needed after #370 gets merged.*

<strike>
Also you need to include the files to the project by adding `EmbeddedResource` to make it work properly in Everest.

https://github.com/EverestAPI/Everest/blob/106d7ed6632e783bf0e4f54bef87d2a25791d0ed/Celeste.Mod.mm/Celeste.Mod.mm.csproj#L182-L185

```diff
   <EmbeddedResource Include="Content\Lua\typebuilder.lua">
     <LogicalName>Content\Lua\typebuilder.lua</LogicalName>
   </EmbeddedResource>
+  <EmbeddedResource Include="Content\Dialog\Japanese.txt">
+    <LogicalName>Content\Dialog\Japanese.txt</LogicalName>
+  </EmbeddedResource>
+  <EmbeddedResource Include="Content\Dialog\Fonts\japanese.fnt">
+    <LogicalName>Content\Dialog\Fonts\japanese.fnt</LogicalName>
+  </EmbeddedResource>
+  <EmbeddedResource Include="Content\Dialog\Fonts\japanese_everest_0.png">
+    <LogicalName>Content\Dialog\Fonts\japanese_everest_0.png</LogicalName>
+  </EmbeddedResource>
+  <EmbeddedResource Include="Content\Dialog\Fonts\japanese_everest_1.png">
+    <LogicalName>Content\Dialog\Fonts\japanese_everest_1.png</LogicalName>
+  </EmbeddedResource>
+  <EmbeddedResource Include="Content\Dialog\Fonts\japanese_everest_2.png">
+    <LogicalName>Content\Dialog\Fonts\japanese_everest_2.png</LogicalName>
+  </EmbeddedResource>
+  <EmbeddedResource Include="Content\Dialog\Fonts\japanese_everest_3.png">
+    <LogicalName>Content\Dialog\Fonts\japanese_everest_3.png</LogicalName>
+  </EmbeddedResource>
+  <EmbeddedResource Include="Content\Dialog\Fonts\japanese_everest_4.png">
+    <LogicalName>Content\Dialog\Fonts\japanese_everest_4.png</LogicalName>
+  </EmbeddedResource>
+  <EmbeddedResource Include="Content\Dialog\Fonts\japanese_everest_5.png">
+    <LogicalName>Content\Dialog\Fonts\japanese_everest_5.png</LogicalName>
+  </EmbeddedResource>
+  <EmbeddedResource Include="Content\Dialog\Fonts\japanese_everest_6.png">
+    <LogicalName>Content\Dialog\Fonts\japanese_everest_6.png</LogicalName>
+  </EmbeddedResource>
 </ItemGroup>
```

</strike>

danesky-jp committed:
Removed duplicated font

danesky-jp commented:
Thank you for valuable information. Following the suggested procedure, I generated font files only for missing characters.

danesky-jp committed:
Updated Japanese.txt

トグルする -> 切り替える

danesky-jp committed:
Merge branch 'EverestAPI:dev' into dev

danesky-jp committed:
Merge branch 'EverestAPI:dev' into dev

danesky-jp committed:
"<" mark is correctly shown in the menu

"<" mark has been missing in the menu in the Japanese environment somehow; "<" has been missing in the font file.
